guides/articles/self-host-hoppscotch-on-your-own-servers.mdx

```diff
@@ -0,0 +1,283 @@
+
+```bash
+#-----------------------Backend Config------------------------------#
+# Prisma Config
+DATABASE_URL=postgresql://username:password@url:5432/dbname # or replace with your database URL
+
+# Auth Tokens Config
+JWT_SECRET=secretcode123
+TOKEN_SALT_COMPLEXITY=10
+MAGIC_LINK_TOKEN_VALIDITY=3
+REFRESH_TOKEN_VALIDITY=604800000 # Default validity is 7 days (604800000 ms) in ms
+ACCESS_TOKEN_VALIDITY=86400000 # Default validity is 1 day (86400000 ms) in ms
+SESSION_SECRET=anothersecretcode123
+
+# Recommended to be true. Set to false if you are using http.
+# Note: Some auth providers may not support http requests and may stop working when set to false.
+ALLOW_SECURE_COOKIES=true
+
+# Hoppscotch App Domain Config
+REDIRECT_URL=http://localhost:3000
+WHITELISTED_ORIGINS=http://localhost:3170,http://localhost:3000,http://localhost:3100
+VITE_ALLOWED_AUTH_PROVIDERS=GOOGLE,GITHUB,MICROSOFT,EMAIL
+
+# Google Auth Config
+GOOGLE_CLIENT_ID=*****
+GOOGLE_CLIENT_SECRET=*****
+GOOGLE_CALLBACK_URL=http://localhost:3170/v1/auth/google/callback
+GOOGLE_SCOPE=email,profile
+
+# Github Auth Config
+GITHUB_CLIENT_ID=*****
+GITHUB_CLIENT_SECRET=****
+GITHUB_CALLBACK_URL=http://localhost:3170/v1/auth/github/callback
+GITHUB_SCOPE=user:email
+
+# Microsoft Auth Config
+MICROSOFT_CLIENT_ID=*****
+MICROSOFT_CLIENT_SECRET=*****
+MICROSOFT_CALLBACK_URL=http://localhost:3170/v1/auth/microsoft/callback
+MICROSOFT_SCOPE=user.read
+MICROSOFT_TENANT=common
+
+# Mailer config
+MAILER_SMTP_ENABLE=true
+MAILER_USE_CUSTOM_CONFIGS=false
+MAILER_ADDRESS_FROM=<from@example.com>
+MAILER_SMTP_URL=smtps://user@domain.com:pass@smtp.domain.com # used if custom mailer configs is false
+# The following are used if custom mailer configs is true
+MAILER_SMTP_HOST=smtp.domain.com
+MAILER_SMTP_PORT=587
+MAILER_SMTP_SECURE=true
+MAILER_SMTP_USER=user@domain.com
+MAILER_SMTP_PASSWORD=pass
+MAILER_TLS_REJECT_UNAUTHORIZED=true
+
+# Rate Limit Config
+RATE_LIMIT_TTL=60 # In seconds
+RATE_LIMIT_MAX=100 # Max requests per IP
+
+#-----------------------Frontend Config------------------------------#
+
+# Base URLs
+VITE_BASE_URL=http://localhost:3000
+VITE_SHORTCODE_BASE_URL=http://localhost:3000
+VITE_ADMIN_URL=http://localhost:3100
+
+# Backend URLs
+VITE_BACKEND_GQL_URL=http://localhost:3170/graphql
+VITE_BACKEND_WS_URL=wss://localhost:3170/graphql
+VITE_BACKEND_API_URL=http://localhost:3170/v1
+
+# Terms Of Service And Privacy Policy Links (Optional)
+VITE_APP_TOS_LINK=https://docs.hoppscotch.io/support/terms
+VITE_APP_PRIVACY_POLICY_LINK=https://docs.hoppscotch.io/support/privacy
+
+# Set to `true` for subpath based access
+ENABLE_SUBPATH_BASED_ACCESS=false
+
+```
+
+### 1. Database Configuration
+
+By default, Hoppscotch ships with a Docker container that has a pre-configured Postgres database. If you prefer to use your own Postgres database, either on the cloud or locally, you’ll need to set up a valid database URL with the correct credentials.
```

Comment:
We dont ship with a pre-configured postgres database, why are we adding this here ?

Reply:
My bad